paasta_tools/check_chronos_jobs.py

```diff
@@ -90,6 +90,23 @@ def sensu_event_for_last_run_state(state):
         return pysensu_yelp.Status.OK
 
 
+def filter_most_recent_jobs_for_service(jobs):
+    most_recent = {}
+    for job in jobs:
+        service, instance = chronos_tools.decompose_job_id(job['name'])
+        if instance not in most_recent:
```

Comment:
The jobs you're passing in here have already been filtered (see the ``chronos_tools.lookup_chronos_jobs`` call on :122 includes ``service`` and ``instance`` filters so you don't need this) and we already have a function for sorting jobs.

Instead, I think the simple version is to remove this new function, and replace the line with ``filtered = chronos_tools.filter_non_temporary_chronos_jobs(matching_jobs)``  with:

```
most_recent_job = [chronos_tools.sort_jobs(matching_jobs)[0]]
```

Note that I'm having to ``make most_recent_job`` a list, because ``last_run_state_for_job`` and ``sensu_message_status_for_jobs`` expect lists rather than individual jobs. This is because we wait until creating the sensu message to check for the case that there is >1 chronos job for a given ``service.instance``, but now that we're sorting the jobs + getting the most recent, this needs to be refactored to be done elsewhere.

I won't block on that though. Ship this and we can sort that after.